src/main/java/org/folio/services/ftp/FtpUploadService.java

```diff
@@ -53,53 +48,39 @@ public boolean isUriValid(String uri) throws URISyntaxException {
     return StringUtils.isEmpty(proto) || proto.equalsIgnoreCase("FTP");
   }
 
-  public Future<String> login(String username, String password) {
-    Promise<String> promise = Promise.promise();
+  @SuppressWarnings("java:S5332")
+  public Future<FTPClient> login(String username, String password) {
+    FTPClient ftpClient = new FTPClient();
+    Promise<FTPClient> promise = Promise.promise();
     ctx.owner().executeBlocking(blockingFeature -> {
-        try {
-          ftp.connect(server, port);
-          logger.info("Connected to {}:{}", server, port);
-          if (ftp.login(username, password)) {
-            blockingFeature.complete(ftp.getReplyString().trim());
-          } else {
-            blockingFeature.fail(new FtpException(ftp.getReplyCode(), ftp.getReplyString().trim()));
-          }
-        } catch (Exception e) {
-          logger.error("Error Connecting", e);
-          blockingFeature.fail(e);
-          disconnect();
+      try {
+        ftpClient.connect(server, port);
+        ftpClient.setDefaultTimeout(30000);
+        ftpClient.addProtocolCommandListener(FTPVertxCommandLogger.getDefListener(logger));
+        ftpClient.setControlKeepAliveTimeout(Duration.ofSeconds(30));
+        ftpClient.setBufferSize(1024 * 1024);
+        ftpClient.setPassiveNatWorkaroundStrategy(new DefaultServerResolver(ftpClient));
+        logger.info("Connected to {}:{}", server, port);
+
+        if (ftpClient.login(username, password)) {
+          blockingFeature.complete(ftpClient);
+        } else {
+          blockingFeature.fail(new FtpException(ftpClient.getReplyCode(), ftpClient.getReplyString().trim()));
         }
-      }, false, asyncResultHandler(promise, "Success login to FTP", "Failed login to FTP"));
+      } catch (Exception e) {
+        logger.error("Error Connecting FTP server {} on port {}", server, port, e);
+        blockingFeature.fail(e);
+        disconnect(ftpClient);
```

Comment:
Do we need to disconnect only for failed cases? If not - we can move disconnect(ftpClient); to finally block

Reply:
we can not create finally block for login , because finally will be always called  and we will be disconnected from ftp client.
finally block present in upload method